### ISSUE TITLE
Matplotlib colormaps for images aren't recognized

### ISSUE THREAD

**Issue by HagaiHargil** [napari/napari#899]
## 🐛 Bug

When trying to use `matplotlib`'s colormaps with napari many of them return a `KeyError` even though the docs says that napari should recognize these names.

## To Reproduce

```python
img = np.array([[1, 0], [0, 1]])
with napari.gui_qt():
    napari.view_image(img, colormap='BuPu')  # KeyError
```

## Expected behavior

napari should support all matplotlib colormaps (as stated), or at least change the docs accordingly.

## Environment

Python 3.7, napari 0.2.10

Full traceback:
```
Traceback (most recent call last):
  File "/data/neural_collision_detection/results/for_article/fig1/code/find_collisions_distribution.py", line 58, in <module>
    napari.view_image(img, colormap='BuPu')  # KeyError
  File "/home/hagai/.conda/envs/ncd/lib/python3.7/site-packages/napari/view_layers.py", line 137, in view_image
    path=path,
  File "/home/hagai/.conda/envs/ncd/lib/python3.7/site-packages/napari/components/viewer_model.py", line 519, in add_image
    visible=visible,
  File "/home/hagai/.conda/envs/ncd/lib/python3.7/site-packages/napari/layers/image/image.py", line 214, in __init__
    self.colormap = colormap
  File "/home/hagai/.conda/envs/ncd/lib/python3.7/site-packages/napari/layers/intensity_mixin.py", line 77, in colormap
    self._cmap = self._colormaps[name]
KeyError: 'BuPu'
```

**Comment by jni** [comment]
We ultimately decided to trim the supported colormaps to a curated list. I guess the easiest fix is to update the docs! And maybe provide a simple code snippet to add matplotlib colormaps. It should be quite easy.

**Comment by jni** [comment]
Specifically, here is how to add a colormap:

https://github.com/napari/napari/blob/master/examples/set_colormaps.py

So you could use

```python
layer.colormap = 'BuPu', vispy.color.Colormap(plt.cm.BuPu(np.linspace(0, 1, 256)))
```

**Comment by HagaiHargil** [comment]
Fair enough, I guess a reference to the available colormaps and to the linked example should be pretty clear.

**Comment by GenevieveBuckley** [comment]
> We ultimately decided to trim the supported colormaps to a curated list.

Was this decision to avoid clutter in the napari GUI dropdown list of colormaps? If so, could we allow support for all matplotlib colormaps "behind the scenes", but keep the dropdown menu to the shorter list of options?

**Comment by sofroniewn** [comment]
Yes @GenevieveBuckley I think this was about not having an overwhelming list in the drop down menu, but maybe we can find a way to support them all and not have them all in the drop down or to add support for better search in the dropdown - see #268 for example. I'm very open to having the full list there or thinking about how to improve this. I will comment in #926 too

**Comment by HagaiHargil** [comment]
I suggest that instead of writing
```python
layer.colormap = 'BuPu', vispy.color.Colormap(plt.cm.BuPu(np.linspace(0, 1, 256)))
```
To change a colormap, we can offer a simpler `layer.colormap = 'BuPu'` option, and if that string is a recognized matplotlib colormap name then napari will automatically create the actual colormap (`vispy.color.Colormap(plt.cm.BuPu(np.linspace(0, 1, 256)))`). If that name isn't in our bank of colormaps then the user will have to enter this tuple manually. It also works nicely with #926 since as mentioned, matplotlib has `*_r` variants for all colormaps.

This change, together with #268, should improve the ergonomics of colormap handling quite considerably.

**Comment by jni** [comment]
@HagaiHargil I like that idea a lot as a first step.

**Comment by sofroniewn** [comment]
> @HagaiHargil I like that idea a lot as a first step.

As long as this approach means we don't have to add matplotlib as an explicit dependency

**Comment by jni** [comment]
Unless memory fails me, I left all of the vendored colormaps in there. Of course, the problem with vendoring: we might now have diverged from matplotlib master, if they added new cmaps in the meantime...

**Comment by HagaiHargil** [comment]
If you really want to make sure napari's always 1:1 with matplotlib's colormaps, you can add matplotlib as a dev dependency and add a test that iterates over both package's colormaps and verifies that all of matplotlib's are included in napari's codebase.

**Comment by GenevieveBuckley** [comment]
> If you really want to make sure napari's always 1:1 with matplotlib's colormaps, you can add matplotlib as a dev dependency and add a test that iterates over both package's colormaps and verifies that all of matplotlib's are included in napari's codebase.

That would be a neat way to handle it, but I don't think matplotlib's colormaps are updated frequently enough for it to be worth the added complexity. As long as the docs here have an accurate list of what is actually available in napari, it's probably less maintenance effort to just update as needed.

**Comment by sofroniewn** [comment]
See discussion in #1575 about extending our colormaps to support matplotlib colormaps

**Comment by liu-ziyang** [comment]
As of today, I am able to run the example given and BuPu is correctly recognized and added as available colormap when configured so, maybe we can close this issue?

**Comment by sofroniewn** [comment]
> As of today, I am able to run the example given and BuPu is correctly recognized and added as available colormap when configured so, maybe we can close this issue?

Yes, turns out this was fixed by #1180 and a [test was added here](https://github.com/napari/napari/blob/v0.3.8/napari/components/_tests/test_viewer_model.py#L40-L41) using a valid colormap string but not one present in our dropdown (in this case `cubehelix`), but we forgot to close this issue. I will close now!

Thanks for the investigation and update @ziyangczi
